src/main/scala/com/sandinh/paho/akka/MqttPubSub.scala

```diff
@@ -18,38 +18,54 @@ object MqttPubSub {
   @inline private def urlDec(s: String) = URLDecoder.decode(s, "utf-8")
 
   def matchTopic(subTopic: String, actualTopic: String): Boolean = {
-    val sub = subTopic.split("/")
-    val actual = actualTopic.split("/")
+
+    val levelSeparator = "/"
+    val multiLevelWildcard = "#"
+    val singleLevelWildcard = "+"
+
+    def prepareTopicsForMatching(actual: String, subscribed: String): (Array[String], Array[String]) = {
+
+      def topicSplit(s: String) =
+        // In Mqtt spec leading or trailing level separators are treated like separating a "empty" level from the following or the preceding, thus when splitting we consider that
+        if(s.endsWith(levelSeparator))
+          s.split(levelSeparator) :+ ""
+        else
+          s.split(levelSeparator)
+
+      (topicSplit(actual),topicSplit(subscribed))
+    }
 
     def doesTopicLevelMatch(actual: String, subscribed: String): Boolean = {
       logger.debug(s"Topic level matching between actual ($actual) subscribed ($subscribed) topic level")
       if( actual == subscribed ){
         logger.debug("matched 1:1")
         true
       }
-      else if ( subscribed == "+" ){
+      else if ( subscribed == singleLevelWildcard ){
         logger.debug("matched single-level wildcard")
         true
       }
-      else if (subscribed == "#"){
+      else if (subscribed == multiLevelWildcard ){
         logger.debug("matched multi-level wildcard")
         true
       }
       else
         false
```

Comment:
this if block seems like a good candidate for a match / case statement, unless you plan to remove the log statements.